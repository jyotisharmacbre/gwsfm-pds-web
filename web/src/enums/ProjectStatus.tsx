enum ProjectStatus {
  InitialCustomerInquiry = 1,
  JA = 2,
  InReview = 3,
  BidLost = 6,
  OnHold = 7,
  OrderReceived = 5,
  JAApproved = 4,
  Live = 9,
  Completed = 10
}

export default ProjectStatus;
