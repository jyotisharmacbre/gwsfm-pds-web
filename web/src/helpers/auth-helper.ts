// @ts-ignore
import authentication from '@kdpw/msal-b2c-react';
import jwtDecode from 'jwt-decode';
import { DecodedToken } from '../models/decodedToken';

export const getDisplayName = () => {
  const decoded: DecodedToken = decodeToken();
  return decoded.name;
};

export const getFirstName = () => {
  const decoded: DecodedToken = decodeToken();
  return decoded.given_name;
};

export const getDisplayEmail = () => {
  const decoded: DecodedToken = decodeToken();
  return decoded.email ?
    decoded.email :
    decoded.emails && decoded.emails.length > 0 ?
      decoded.emails[0] :
      '';
};

export const logOut = () => {
  authentication.signOut();
}

function decodeToken() {
  const token = authentication.getAccessToken();
  const decoded: DecodedToken = jwtDecode(token);
  return decoded;
}

