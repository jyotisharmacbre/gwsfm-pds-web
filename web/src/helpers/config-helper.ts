const appConfig = () => {
    const config: any = (window as any).Config;
    if(config && config.REACT_APP_AUTH_INSTANCE==="#{AUTH_INSTANCE}#"){
      return {
        ...process.env
      }
    }
    return {
      ...config
    }
  };
  
  export default appConfig;