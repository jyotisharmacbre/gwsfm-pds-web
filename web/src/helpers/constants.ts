export const selectionButtons = [
  {
    title: 'BUTTON_YES',
    value: true
  },
  {
    title: 'BUTTON_NO',
    value: false
  }
];

export enum Notify {
  none,
  success,
  error
}
