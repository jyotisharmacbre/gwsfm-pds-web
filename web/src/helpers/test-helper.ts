import checkPropTypes from 'check-prop-types';

export const findByTestAtrr = (component: any, attr: string) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};

export const checkProps = (component: any, expectedProps: any) => {
  const propsErr = checkPropTypes(
    component.propTypes,
    expectedProps,
    'props',
    component.name
  );
  return propsErr;
};
