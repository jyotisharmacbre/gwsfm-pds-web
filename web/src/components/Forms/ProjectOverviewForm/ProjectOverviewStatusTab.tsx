import React from 'react';
import { faLightbulb, faPencilAlt } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';


import { injectIntl, FormattedMessage } from 'react-intl';

import { confirmAlert } from '../../Popup/CustomModalPopup';
import IReactIntl from '../../../Translations/IReactIntl';
import ProjectStatus from '../../../enums/ProjectStatus';
import { activeMocks } from 'nock/types';

interface IProps {
    status: number;
    statusName: string;
    onReactivate: () => void;
    handleBidLost: () => void;
    handleOnHold: () => void;
}




const ProjectOverviewStatusTab: React.FC<IProps & IReactIntl> = props => {
    const handleToggleStatusTab = () => {
        var element: any = document.getElementById('statusTab');
        if (element != null) {
            var isClassExists = element.classList.contains('show');
            if (isClassExists) {
                element.classList.add('hide');
                element.classList.remove('show');
            } else {
                element.classList.remove('hide');
                element.classList.add('show');
            }
        }
    }
    const activateStatus = () => {
    var element: any = document.getElementById('activaty_btn');
    if (element != null) {
        var isClassExists = element.classList.contains('active');
        if (isClassExists) {
            //element.classList.add('');
            element.classList.remove('active');
        } else {
            //element.classList.remove('active');
            element.classList.add('active');
        }
    }
};
    
    return (

        <div className="col-md-6 mt-4 mt-lg-0 d-flex justify-content-start justify-content-lg-end">
            <div id="activaty_btn" className="activate_status_box" onClick={() => activateStatus()}>
                {(props.status == ProjectStatus.BidLost || props.status == ProjectStatus.OnHold) ? <div className="status-dropdown-btn toggle mrgnrght10 mrgnt5">
                    <span>

                        <button type="button" id="reactivateButton" data-test="activateButton"
                            onClick={() =>
                                confirmAlert({
                                    intl: props.intl,
                                    titleKey: "TITLE_CONFIRMATION",
                                    contentKey: "MESSAGE_PROJECT_REACTIVATE",
                                    handleConfirm: () => props.onReactivate()
                                })}>
                            <FontAwesomeIcon className="active mrgnrght10" icon={faLightbulb} />ACTIVATE
                                            </button>
                    </span>
                </div> : null}
                <div className="status_btn">
                    <div className="status-dropdown">
                        <div className="status-dropdown-btn toggle">

                            <span data-test="toggleStatusTab" id="toggleStatusTab" className={(props.status == ProjectStatus.BidLost || props.status == ProjectStatus.OnHold) ? "dropdown-placeholder link_disabled p-0" : "dropdown-placeholder p-0"} onClick={() => handleToggleStatusTab()}>Status:&nbsp; <strong>{props.statusName}
                            </strong>{(props.status != ProjectStatus.BidLost && props.status != ProjectStatus.OnHold) ? <FontAwesomeIcon className="active mrgnlft10" icon={faPencilAlt} /> : null}
                            </span>
                        </div>
                        {(props.status != ProjectStatus.BidLost && props.status != ProjectStatus.OnHold) ? <div className="status-dropdown-menu hide status-hidden toggle-list" data-test="statusTab" id="statusTab">
                            <p>Change Status to</p>
                            <ul className="status-dropdown-list status-scrollable">
                                <li data-test="bidlost" className={(props.status == 4) ? "status-dropdown-item mrgnlft10 status-selected link_disabled" : "status-dropdown-item mrgnlft10"} onClick={() =>
                                    confirmAlert({
                                        intl: props.intl,
                                        titleKey: "TITLE_CONFIRMATION",
                                        contentKey: "MESSAGE_PROJECT_STATUS_CHANGE",
                                        handleConfirm: () => props.handleBidLost()
                                    })
                                }>
                                    <a title="Bid Lost">Bid
                                                                    Lost</a></li>
                                <li data-test="onhold" className={(props.status == 6) ? "status-dropdown-item mrgnlft10 status-selected link_disabled" : "status-dropdown-item mrgnlft10 "} onClick={() =>
                                    confirmAlert({
                                        intl: props.intl,
                                        titleKey: "TITLE_CONFIRMATION",
                                        contentKey: "MESSAGE_PROJECT_STATUS_CHANGE",
                                        handleConfirm: () => props.handleOnHold()
                                    })}>
                                    <a
                                        title="On Hold">On
                                                                    Hold</a></li>

                            </ul>

                        </div> : null}
                    </div>
                </div>
            </div>
        </div>

    )

}

export default injectIntl(ProjectOverviewStatusTab);
