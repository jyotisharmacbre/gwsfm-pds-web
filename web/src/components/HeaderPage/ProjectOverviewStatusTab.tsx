import React from 'react';
import { faLightbulb,faPencilAlt } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { confirmAlert } from '../Popup/CustomModalPopup';
import { injectIntl, FormattedMessage } from 'react-intl';
import IReactIntl from '../../Translations/IReactIntl';
import ProjectStatus from '../../enums/ProjectStatus';

interface IProps
{
    status:number;
    statusName:string;
    onReactivate:()=>void;
    handleBidLost:()=>void;
    handleOnHold:()=>void;
}

const ProjectOverviewStatusTab: React.FC<IProps&IReactIntl> = props => {  
    const handleToggleStatusTab=()=>{
        var element: any = document.getElementById('statusTab');
        if (element != null) {
          var isClassExists = element.classList.contains('show');
          if (isClassExists) {
            element.classList.add('hide');
            element.classList.remove('show');
          } else {
            element.classList.remove('hide');
            element.classList.add('show');
          }
        }
    }
    return(
      
             <div className="col-md-6 float-right">
                                        <div className="activate_status_box">
                                        {(props.status==ProjectStatus.bidlost || props.status==ProjectStatus.onhold)? <div className="status-dropdown-btn toggle mrgnrght10 mrgnt5">
                                                        <span>
                                                       
                                                        <button type="button"  id="reactivateButton" data-test="activateButton" 
                                                        onClick={()=>
                                                        confirmAlert({
                                                        intl:props.intl,
                                                        titleKey:"TITLE_CONFIRMATION",
                                                        contentKey:"MESSAGE_PROJECT_REACTIVATE",
                                                        handleConfirm:()=>props.onReactivate()
                                                        })}>
                                            <FontAwesomeIcon className="active mrgnrght10" icon={faLightbulb} />ACTIVATE
                                            </button>
                                                        </span>
                                           </div>:
                                            <div className="status_btn">
                                                <div className="status-dropdown">
                                                    <div className="status-dropdown-btn toggle">
                                                       
                                                        <span data-test="toggleStatusTab" id="toggleStatusTab" className="dropdown-placeholder" onClick={()=>handleToggleStatusTab()}>Status:&nbsp; <strong>{props.statusName}
                                                        (<FormattedMessage id="LABEL_LIVE"/>)</strong><FontAwesomeIcon className="active mrgnlft10" icon={faPencilAlt} />
                                                        </span>
                                                    </div>
                                                    <div className="status-dropdown-menu hide status-hidden toggle-list" data-test="statusTab" id="statusTab">
                                                        <p>Change Status to</p>
                                                        <ul className="status-dropdown-list status-scrollable">
                                                            <li data-test="bidlost"  className={(props.status==4)?"status-dropdown-item mrgnlft10 status-selected link_disabled":"status-dropdown-item mrgnlft10"} onClick={()=>
                                                                confirmAlert({
                                                                    intl:props.intl,
                                                                    titleKey:"TITLE_CONFIRMATION",
                                                                    contentKey:"MESSAGE_PROJECT_STATUS_CHANGE",
                                                                    handleConfirm:()=>props.handleBidLost()
                                                                  })
                                                                }> 
                                                                <a title="Bid Lost">Bid
                                                                    Lost</a></li>
                                                            <li data-test="onhold" className={(props.status==6)?"status-dropdown-item mrgnlft10 status-selected link_disabled":"status-dropdown-item mrgnlft10 "} onClick={()=>
                                                            confirmAlert({
                                                            intl:props.intl,
                                                            titleKey:"TITLE_CONFIRMATION",
                                                            contentKey:"MESSAGE_PROJECT_STATUS_CHANGE",
                                                            handleConfirm:()=>props.handleOnHold()
                                                            })}>
                                                                    <a
                                                                    title="On Hold">On
                                                                    Hold</a></li>

                                                        </ul>
                                                       
                                                    </div>
                                                </div>
                                            </div>}
                                        </div>
                                    </div>
    ) 
   
}

export default injectIntl(ProjectOverviewStatusTab);
