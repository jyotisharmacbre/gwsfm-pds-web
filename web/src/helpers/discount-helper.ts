import DiscountType from '../enums/DiscountType';
import { restrictMinusAndAllowDecimalForMaxRangeHundred, restrictMinusAndAllowDecimal } from '../helpers/utility-helper';

const normalizeClientDiscount = (discountType: DiscountType, discountValue) => {
    let updatedDiscount;
    if (discountType === DiscountType.Percent) {
        updatedDiscount = restrictMinusAndAllowDecimalForMaxRangeHundred(discountValue);
    }
    else updatedDiscount = restrictMinusAndAllowDecimal(discountValue);
    return updatedDiscount;
}

export default normalizeClientDiscount;