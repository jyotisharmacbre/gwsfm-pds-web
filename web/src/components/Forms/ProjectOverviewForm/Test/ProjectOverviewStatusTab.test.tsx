

import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { IntlProvider } from 'react-intl';
import configureStore from 'redux-mock-store';
import translations from '../../../../Translations/translation';
import ProjectOverviewStatusTab from '../ProjectOverviewStatusTab';
import { findByTestAtrr } from '../../../../helpers/test-helper';
import ProjectStatus from '../../../../enums/ProjectStatus';
const props: any = {
  status: 1,
  statusName: "test",
  onReactivate: jest.fn(),
  handleBidLost: jest.fn(),
  handleOnHold: jest.fn(),
  handleOrderReceived: jest.fn()
};
let wrapper;
const mockStore = configureStore([]);
const mountComponent = (props) => {
  wrapper = mount(
    <Provider store={mockStore({})}>
      <IntlProvider locale="en" messages={translations['en'].messages}>
        <ProjectOverviewStatusTab {...props} />
      </IntlProvider>
    </Provider>
  );
}
describe('PdsFormButton Renders', () => {

  beforeEach(() => {
    mountComponent(props);
  });
  it('Defines the component', () => {
    expect(wrapper).toBeDefined();
  });
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should not display reactivate button if status is not bid lost or on hold', () => {

    let button = findByTestAtrr(wrapper, "activateButton").first();
    expect(button.length).toBe(0);
  });
  it('should  display reactivate button if status is bid lost or on hold', () => {
    const testProps: any = {
      status: ProjectStatus.BidLost,
      statusName: "test",
      onReactivate: jest.fn(),
      handleBidLost: jest.fn(),
      handleOnHold: jest.fn()
    };
    mountComponent(testProps);
    let button = findByTestAtrr(wrapper, "activateButton").first();
    expect(button.length).toBe(1);
  });
  it('should display status tab on click of status edit icon', () => {
    let container = findByTestAtrr(wrapper, "toggleStatusTab").first();
    container.simulate("click");
    let button = findByTestAtrr(wrapper, "statusTab").first();
    expect(button.length).toBe(1);
    expect(button.hasClass('hide')).toBe(true);
  });
  it('should trigger bid lost action handler on click of bid lost', () => {
    let container = findByTestAtrr(wrapper, "bidlost").first();
    container.simulate("click");
    expect(container.invoke.call.length).toBe(1);
  });
  it('should trigger on hold action handler on click of on hold', () => {
    let container = findByTestAtrr(wrapper, "onhold").first();
    container.simulate("click");
    expect(container.invoke.call.length).toBe(1);
  });
  it('should trigger reactivate action handler on click of active button', () => {
    let container = findByTestAtrr(wrapper, "toggleStatusTab").first();
    container.simulate("click");
    expect(container.invoke.call.length).toBe(1);
  });
  it('should display reactivate action and status bar in disable state if status is bidlost or onhold', () => {
    const testProps: any = {
      status: ProjectStatus.BidLost,
      statusName: "test",
      onReactivate: jest.fn(),
      handleBidLost: jest.fn(),
      handleOnHold: jest.fn()
    };
    mountComponent(testProps);
    let button = findByTestAtrr(wrapper, "activateButton").first();
    let container = findByTestAtrr(wrapper, "toggleStatusTab").first();
    expect(button.length).toBe(1);
    expect(container.hasClass("link_disabled")).toBe(true);
  });
  it('should display orderReceived option in dropdown if status is JAApproved', () => {
    const testProps: any = {
      status: ProjectStatus.JAApproved,
      statusName: "test",
      onReactivate: jest.fn(),
      handleBidLost: jest.fn(),
      handleOnHold: jest.fn()
    };
    mountComponent(testProps);
    let button = findByTestAtrr(wrapper, "orderReceived").first();
    expect(button.length).toBe(1);
  });
  it('should trigger orderReceived  event on click of dropdown action', () => {
    const testProps: any = {
      status: ProjectStatus.JAApproved,
      statusName: "test",
      onReactivate: jest.fn(),
      handleBidLost: jest.fn(),
      handleOnHold: jest.fn()
    };
    mountComponent(testProps);
    let container = findByTestAtrr(wrapper, "orderReceived").first();
    container.simulate("click");
    expect(container.invoke.call.length).toBe(1);
  });

  it('should not display reactivate action and status bar in disable state if status is InReview', () => {
    const testProps: any = {
      status: ProjectStatus.InReview,
      statusName: "test",
      onReactivate: jest.fn(),
      handleBidLost: jest.fn(),
      handleOnHold: jest.fn()
    };
    mountComponent(testProps);
    let container = findByTestAtrr(wrapper, "toggleStatusTab").first();
    container.simulate("click");
    //toggeling again
    container.simulate("click");
    expect(container.invoke.call.length).toBe(1);
  });


  it('should  render project live status in dropdown and should hide bid lost and on hold when project status is OrderRecived', () => {
    const testProps: any = {
      status: ProjectStatus.OrderReceived,
      statusName: "test"
    };
    mountComponent(testProps);
    let field = findByTestAtrr(wrapper, "live").first();
    expect(field.length).toBe(1);
  });

  it('should  render project complete status in dropdown and should hide bid lost and on hold when project status is live', () => {
    const testProps: any = {
      status: ProjectStatus.Live,
      statusName: "test"
    };
    mountComponent(testProps);
    let field = findByTestAtrr(wrapper, "completed").first();
    expect(field.length).toBe(1);
  });

  it('should trigger handleProjectLive  event on click of dropdown action', () => {
    const testProps: any = {
      status: ProjectStatus.OrderReceived,
      statusName: "test",
      handleProjectLive: jest.fn()
    };
    mountComponent(testProps);
    const container = findByTestAtrr(wrapper, "live").first();
    container.simulate("click");
    expect(container.invoke.call.length).toBe(1);
  });

  it('should trigger handleProjectCompleted  event on click of dropdown action', () => {
    const testProps: any = {
      status: ProjectStatus.Live,
      statusName: "test",
      handleProjectLive: jest.fn()
    };
    mountComponent(testProps);
    const container = findByTestAtrr(wrapper, "completed").first();
    container.simulate("click");
    expect(container.invoke.call.length).toBe(1);
  });
  
  it('should hide status tab on click outside of status tab', () => {
    let container = findByTestAtrr(wrapper, "toggleStatusTab").first();
    container.simulate("click");
    let field = findByTestAtrr(wrapper, "bidlost").first();
    field.simulate("click");
    let button = findByTestAtrr(wrapper, "statusTab").first();
    expect(button.hasClass('hide')).toBe(true);
  });

});




