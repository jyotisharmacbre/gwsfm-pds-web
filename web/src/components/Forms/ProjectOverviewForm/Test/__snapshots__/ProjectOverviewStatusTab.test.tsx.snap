// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PdsFormButton Renders should match the snapshot 1`] = `ReactWrapper {}`;
